File: src/001.Array/485.MaxConsecutiveOnes/485.最大连续-1-的个数.go
/*
 * @lc app=leetcode.cn id=485 lang=golang
 *
 * [485] 最大连续 1 的个数
 */

// @lc code=start
func findMaxConsecutiveOnes(nums []int) int {
	maxOnesLen := 0
	count := 0

	for _, v := range nums {
		if v == 1 {
			count++
		} else {
			maxOnesLen = max(maxOnesLen, count)
			count = 0
		}
	}

	maxOnesLen = max(maxOnesLen, count)

	return maxOnesLen
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// @lc code=end

File: src/001.Array/485.MaxConsecutiveOnes/485.最大连续-1-的个数.js
/*
 * @lc app=leetcode.cn id=485 lang=javascript
 *
 * [485] 最大连续 1 的个数
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
	let maxOnesLen = 0;

	if (!Array.isArray(nums)) return maxOnesLen;

	const strNums = nums.join('');
	const arrOnes = strNums.split('0');

	arrOnes.forEach((i) => {
        if (i.length > maxOnesLen) {
            maxOnesLen = i.length;
        }
    });

    return maxOnesLen;
};
// @lc code=end
